File: Python3/프로그래머스/2/42586. 기능개발/기능개발.py
import math
def solution(progresses, speeds):
    answer, idx, n = [], 0, 1
    
    rest = 0
    while idx < len(speeds):
        rest = math.ceil((100 - progresses[idx]) / speeds[idx])
        # print("첫번째 값이 채워질 때까지 남은 시간:", rest)
        n = 0
        
        for i in range(idx, len(speeds)):
            progresses[i] += speeds[i] * rest
            if i == idx and progresses[i] >= 100:
                idx += 1
                n += 1
        # print("progresses:", progresses)
        answer.append(n)
        
    
    return answer

File: C++/프로그래머스/2/42586. 기능개발/기능개발.cpp
#include <vector>
#include <cmath>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> progresses, vector<int> speeds) {
    vector<int> answer;
    int idx = 0, time = 0, n = 0;
    
    while(idx < speeds.size()) {
        n = 0;
        time = ceil((100 - progresses[idx]) / float(speeds[idx]));
        // cout << "현재 추가된 시간: " << time << endl;
        
        for(int i = idx; i < speeds.size(); i++) {
            progresses[i] += time * speeds[i];
            if(i == idx && progresses[i] >= 100) {
                n++;
                idx++;
            }
        }
        
        // cout << "[";
        // for(auto& proc : progresses) {
        //     cout << proc << " ";
        // }
        // cout << "]" << endl;
        answer.push_back(n);
    }
    
    return answer;
}